src/components/LetterSelection/index.jsx

```diff
@@ -1,9 +1,130 @@
+
+    case types.addLetter: // checks if clicked letter's index is already in clickedLetterIndexes of the letterState and if letterCounter is lower than the word's length, then replaces empty strings in clickedLetters array with letter, "pushes" index of letter to clickedLetterIndexes, and increments the counter
+      if (
+        !state.clickedLetterIndexes.includes(action.payload.index) &&
+        state.letterCounter < state.clickedLetters.length
+      ) {
+        return {
+          ...state,
+          letterCounter: state.letterCounter + 1,
+          clickedLetters: state.clickedLetters.map((letter, index) => {
+            if (index === state.letterCounter) {
+              letter = action.payload.letter;
+            }
+            return letter;
+          }),
+          clickedLetterIndexes: [
+            ...state.clickedLetterIndexes,
+            action.payload.index,
+          ],
+        };
+      } else {
+        return state;
+      }
+
+    case types.removeLetter:
+      const clickedLetterIndexesPopped = state.clickedLetterIndexes.slice(
+        0,
+        -1
+      );
+      return {
+        ...state,
+        letterCounter: state.letterCounter === 0 ? 0 : state.letterCounter - 1,
+        clickedLetters: state.clickedLetters.map((letter, index) => {
+          if (index + 1 === state.letterCounter) {
+            letter = '';
+          }
+          return letter;
+        }),
+        clickedLetterIndexes: clickedLetterIndexesPopped,
+      };
+
+    case types.initialize: // initalize the state with resetting letterCounter, new selectable letters, and new blank strings
+      return {
+        ...state,
+        letterCounter: 0,
+        selectableLetters: makeSelectableLetters(action.payload),
+        clickedLetters: makeBlankLetters(action.payload),
+        clickedLetterIndexes: [],
+      };
+
+    default:
+      return state;
+  }
+};
+
+const initialLetterState = {
+  letterCounter: 0,
+  selectableLetters: [],
+  clickedLetters: [], // keeps track of letters that were clicked
+  clickedLetterIndexes: [], // keeps track of letter indexes that were clicked
+};
+
+export default function LetterSelection({ word, dispatchGame, typesGame }) {
+  const [letterState, dispatch] = useReducer(reducer, initialLetterState);
+
+  useEffect(() => {
+    dispatch({
+      type: types.initialize,
+      payload: word,
+    });
+  }, [word]);
 
-export default function LetterSelection() {
   return (
     <>
-      <h2>Letters to choose</h2>
+      {letterState.clickedLetters.map((clickedLetter, index) => {
+        return <div key={index}>{clickedLetter}</div>;
+      })}
+
+      {/* Buttons to select the letters */}
+      {letterState.selectableLetters.map((selectableLetter, index) => {
+        return (
+          <button
+            key={index}
+            onClick={() =>
+              dispatch({
+                type: types.addLetter,
+                payload: { index: index, letter: selectableLetter },
```

Comment:
Just so I understand: you're not asking me to remove the clickedLetters array property from the reducer state, but rather asking me to not send the letter payload since I can use the index of it? (i.e. `selectableLetters[index]`)?